src/tests/conformance/messages.rs

```diff
@@ -184,6 +184,129 @@ async fn ignores_unsolicited_responses() {
+    //          sometimes even spuriously failing the test completely.
+
+    let (zig, node_meta) = read_config_file();
+
+    // create tcp listeners for peer set (port is only assigned on tcp bind)
+    let mut listeners = Vec::new();
+    for _ in 0u8..5 {
+        listeners.push(TcpListener::bind(zig.new_local_addr()).await.unwrap());
+    }
+
+    // get list of peer addresses
+    let peer_addresses = listeners
+        .iter()
+        .map(|listener| listener.local_addr().unwrap())
+        .collect::<Vec<_>>();
+
+    // start the node
+    let mut node = Node::new(node_meta);
+    node.start_waits_for_connection(zig.new_local_addr())
+        // .log_to_stdout(true)
+        .start()
+        .await;
+
+    // start peer listeners which "pass" once they've accepted a connection, and
+    // "fail" if the timeout expires. Timeout must be quite long, seems to take around
+    // 20-60 seconds for zebra.
+    let mut peer_handles = Vec::with_capacity(listeners.len());
+    for listener in listeners {
+        peer_handles.push(tokio::time::timeout(
+            tokio::time::Duration::from_secs(120),
+            tokio::spawn(async move {
+                println!(
+                    "{:?}: waiting for connection",
+                    listener.local_addr().unwrap()
+                );
+                listener.accept().await.unwrap();
+                println!("{:?}: connected!", listener.local_addr().unwrap());
+            }),
+        ));
+    }
+
+    // connect to the node main
+    let mut stream = initiate_handshake(node.addr()).await.unwrap();
+
+    // wait for the `GetAddr`, filter out all other queries.
+    let filter = MessageFilter::with_all_auto_reply()
+        .enable_logging()
+        .with_getaddr_filter(crate::protocol::message::Filter::Disabled);
+
+    // reply with list of peer addresses
+    match filter.read_from_stream(&mut stream).await.unwrap() {
+        Message::GetAddr => {
+            let peers = peer_addresses
+                .iter()
+                .map(|addr| NetworkAddr::new(*addr))
+                .collect::<Vec<_>>();
+
+            Message::Addr(Addr::new(peers))
+                .write_to_stream(&mut stream)
+                .await
+                .unwrap();
+        }
+        message => panic!("Expected Message::GetAddr, but got {:?}", message),
+    }
+
+    // turn waiting for peer futures into a single future
+    let wait_for_peers = tokio::spawn(async move {
+        for handle in peer_handles {
+            handle.await.unwrap().unwrap();
+        }
+    });
+
+    // We need to keep responding to ping requests on the main connection,
+    // otherwise it may get marked as unreliable (and the peer list gets ignored).
+    //
+    // Without this, zebra takes forever to connect and spuriously fails as well.
+    // TBC - this is all speculation.
+    let main_responder = tokio::spawn(async move {
```

Comment:
Great! This made a huge difference for me locally: test runs in under one second.